File: maxProfit.c
int maxProfit(int* prices, int pricesSize){
    int maxPro = 0;
    int min = prices[0];    //先让0位置为最小
    for(int i=1; i<pricesSize; i++){
        if(prices[i] > min){
            maxPro = fmax(maxPro, prices[i] - min);   //如果后面比最小大，求差，与现有最大比取最大值
        }else if(prices[i] < min){
            min = prices[i];    //如果后面比最小还小，那么替换最小值，从这里继续算起
        }
    }
    return maxPro;
}

File: maxProfit.kt
/*class Solution {
    fun maxProfit(prices: IntArray): Int {
        var minprice = Int.MAX_VALUE
        var maxprofit = 0
        for (price in prices) {
            if (price < minprice) {
                minprice = price
            } else if (price - minprice > maxprofit) {
                maxprofit = price - minprice
            }
        }
        return maxprofit
    }
}*/

class Solution {
    fun maxProfit(prices: IntArray): Int {
         var minprice = Int.MAX_VALUE
        var maxprofit = 0
        for ( i in 0 until prices.count()) {
            if (prices[i] < minprice) {
                minprice = prices[i]
            } else if (prices[i] - minprice > maxprofit) {
                maxprofit = prices[i] - minprice
            }
        }
        return maxprofit
    }
}


